src/api.ts

```diff
@@ -15,11 +15,14 @@ router.post('/', async (req, res) => {
   if (!method) return rpcError(res, 400, 'missing method', id);
   try {
     let result: any = {};
+    if (!Array.isArray(params)) return rpcError(res, 400, 'params must be an array of string', id);
+
     if (method === 'lookup_addresses') result = await lookupAddresses(params);
+    else if (method === 'resolve_name') result = await resolveName(params[0]);
     else return rpcError(res, 400, 'invalid method', id);
 
-    if (result.error) return rpcError(res, result.code || 500, result.error, id);
-    return rpcSuccess(res, result, id);
+    if (result?.error) return rpcError(res, result.code || 500, result.error, id);
+    return rpcSuccess(res, { address: result[params[0]] }, id);
```

Comment:
I feel accepting array may confuse people, as they may think they can pass multiple values, but it will always only resolve the first one.

Would it be better if we just add support for multiple values ?

Reply:
Yea, feel like better to return an error if more than one name is passed, but if it is easy, we can add a limit up to 5 names for now

Reply:
Ok, will proceed to a refactoring to add support to multiple handle resolbver